### ISSUE TITLE
Examples requiring  gi-haskell-base 0.23 (invalid package?)

### ISSUE THREAD

**Issue by santiweight** [haskell-gi/gi-gtk-examples#6]
I  just  tried compiling some of the examples and it  seems they  require a non-existent version of  haskell-gi-base, at least according to  Hackage. The latest version of haskell-gi-base appears to be 0.22.2 but  the  examples import from 0.23.*. Is this  an error or can I find this package elsewhere? Neither nightly  or lts seemed to accept the packages listed unless I  change the requirements to  0.22.2.

**Comment by garetxe** [comment]
Indeed. The examples in the git repo depend on the versions of `haskell-gi` and `haskell-gi-base` in the repo (`0.23.*`), which I have not yet released to hackage (it should happen soon, whenever I find some time). There is a good chance that many of the examples work with older versions (perhaps after minor modifications), so it's worth just trying to relax the lower bound in the `.cabal` file.

Alternatively, if you want to build the git repo, you can do:
```
pushd bindings
cabal new-run genBuildInfo $(PKGS.sh)
popd
cabal new-run <the example you want>
```
You might need to install some devel packages to be able to compile all bindings. Alternatively, edit `bindings/PKGS.sh` and `cabal.project` to remove the bindings you are not interested in.

**Comment by garetxe** [comment]
Actually, since you are probably running this from OSX, perhaps better replacing `PKGS.sh` with `OSX-PKGS.sh` above, and `cabal new-run ...` with `cabal new-run --project-file=osx.cabal.project ...`. (After adding the examples you are interested in building to `osx.cabal.project`.)
